src/test/java/teammates/sqllogic/core/AccountsLogicTest.java

```diff
@@ -0,0 +1,170 @@
+package teammates.sqllogic.core;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.NotificationStyle;
+import teammates.common.datatransfer.NotificationTargetUser;
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.exception.InvalidParametersException;
+import teammates.storage.sqlapi.AccountsDb;
+import teammates.storage.sqlentity.Account;
+import teammates.storage.sqlentity.Notification;
+import teammates.storage.sqlentity.ReadNotification;
+import teammates.test.BaseTestCase;
+
+/**
+ * SUT: {@link AccountsLogic}.
+ */
+public class AccountsLogicTest extends BaseTestCase {
+
+    private AccountsLogic accountsLogic = AccountsLogic.inst();
+
+    private AccountsDb accountsDb;
+
+    private NotificationsLogic notificationsLogic;
+
+    @BeforeMethod
+    public void setUpMethod() {
+        accountsDb = mock(AccountsDb.class);
+        notificationsLogic = mock(NotificationsLogic.class);
+        accountsLogic.initLogicDependencies(accountsDb, notificationsLogic);
+    }
+
+    @Test
+    public void testUpdateReadNotifications_success() throws InvalidParametersException, EntityDoesNotExistException {
```

Comment:
This particular unit test is trying to do too much. Should split into separate tests cases for each behaviour.

Issue when you combine these is that
1) They take forever to write
2) Hard to localise any faults, especially since we are using these mainly as regression tests.